src/main/java/net/minecraftforge/registries/RegistryObject.java

```diff
@@ -111,15 +115,25 @@ private RegistryObject() {
         ObjectHolderRegistry.addHandler(new Consumer<Predicate<ResourceLocation>>()
         {
             private IForgeRegistry<V> registry;
+            private boolean invalidRegistry = false;
 
             @Override
             public void accept(Predicate<ResourceLocation> pred)
             {
+                if (invalidRegistry)
+                    return;
                 if (registry == null)
                 {
                     this.registry = RegistryManager.ACTIVE.getRegistry(baseType);
                     if (registry == null)
-                        throw new IllegalStateException("Unable to find registry for type " + baseType.getName() + " for mod \"" + modid + "\". Check the 'caused by' to see further stack.", callerStack);
+                    {
+                        invalidRegistry = true;
+                        LOGGER.error(
+                                "Unable to find registry for type " + baseType.getName() + " for mod \"" + modid + "\". Check the 'caused by' to see further stack.",
+                                callerStack
+                        );
```

Comment:
Use logger parameter substitution, and move this all to one line:
```suggestion
                        LOGGER.error("Unable to find registry for type {} for mod \"{}\". Check the 'caused by' to see further stack.", baseType.getName(), modid, callerStack);
```